src/plasoscaffolder/frontend/controller/sqlite_controller.py

```diff
@@ -75,52 +87,180 @@ def PluginName(self, ctx: click.core.Context, param: click.core.Option,
+            unused_param: click.core.Option,
             value: str) -> str:
     """The events of the plugin
 
     Args:
-      ctx (click.core.Context): the click context (automatically given via
+      unused_ctx (click.core.Context): the click context (automatically given
+      via
       callback)
-      param (click.core.Option): the click command (automatically given via
+      unused_param (click.core.Option): the click command (automatically
+      given via
       callback)
       value (str): the source path (automatically given via callback)
 
     Returns:
       str: the events of the plugin
     """
-    self._events = value.title().split()
-    return self._events
+    event_model_list = []
+    for event_name in value.title().split():
+      event_model_list.append(self._CreateEventModelWithUserInput(event_name))
+    self._events = event_model_list
+
+    return event_model_list
+
+  def SQLQuery(self, unused_ctx: click.core.Context,
+               unused_param: click.core.Option,
+               value: str) -> str:
+    """The events of the plugin
+
+    Args:
+      unused_ctx (click.core.Context): the click context (automatically given
+      via
+      callback)
+      unused_param (click.core.Option): the click command (automatically
+      given via
+      callback)
+      value (str): the sql query (automatically given via callback)
+
+    Returns:
+      str: the sql query
+    """
+
+    verbose = value
+    add_more_queries = True
+    sql_query_list = []
+    while add_more_queries:
+      sql_query = self._output_handler.PromptInfo(
+          text='Please write your SQL script for the plugin')
+      query_model = self._CreateSQLQueryModelWithUserInput(sql_query, verbose,
+                                                           self._query_execution)
+      if query_model is not None:
+        sql_query_list.append(query_model)
+        add_more_queries = self._output_handler.Confirm(
+            text='Do you want to add another query?',
+            abort=False, default=True)
+
+    self._sql_query = sql_query_list
+    return sql_query_list
+
+  def _CreateSQLQueryModelWithUserInput(
+      self,
+      query: str, with_examples: bool,
+      query_execution: base_sql_query_execution.BaseSQLQueryExecution()
+  ) -> sql_query_model.SQLQueryModel:
+    """Asks the user information about the sql query
+  
+    Args:
+      query (str): the sql query
+      with_examples (bool): if the user wants examples for the given query
+  
+    Returns:
+      (sql_query_model.SQLQueryModel) a sql query model
+    """
+    query_data = self._plugin_helper.RunSQLQuery(query, query_execution)
+
+    if query_data.has_error:
+      self._output_handler.PrintError('The SQLQuery has an Error.')
+      self._output_handler.PrintError(str(query_data.error_message))
+      return None
+
+    else:
+      if with_examples:
+        length = len(query_data.data)
+        if length is 0 :
```

Comment:
use `length == 0` only use is for objects e.g. `length is None`

Also see: http://stackoverflow.com/questions/1504717/why-does-comparing-strings-in-python-using-either-or-is-sometimes-produce